File: panel/src/features/auth/data/models/AuthUserModel.ts
import type { AuthUser } from '../../domain/entities/AuthUser';

export class AuthUserModel {
  static fromJson(json: Record<string, unknown>): AuthUser {
    return {
      id: String(json.id),
      email: String(json.email),
      firstName: String(json.firstName),
      lastName: String(json.lastName),
      role: String(json.role),
      status: String(json.status),
      avatarUrl: json.avatarUrl ? String(json.avatarUrl) : null,
    };
  }
}

File: app/lib/features/auth/data/models/auth_user_model.dart
import '../../domain/entities/auth_user.dart';

class AuthUserModel extends AuthUser {
  const AuthUserModel({
    required super.id,
    required super.email,
    required super.firstName,
    required super.lastName,
    required super.role,
    required super.status,
    super.avatarUrl,
  });

  factory AuthUserModel.fromJson(Map<String, dynamic> json) {
    return AuthUserModel(
      id: json['id'] as String,
      email: json['email'] as String,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      role: json['role'] as String,
      status: json['status'] as String,
      avatarUrl: json['avatarUrl'] as String?,
    );
  }
}